facebook/delphiFacebook/R/aggregate.R

```diff
@@ -181,7 +181,7 @@ summarize_indicators_day <- function(day_df, indicators, target_day, geo_level,
 
       # Copy only columns we're using.
       select_cols <- c(metric, var_weight, "weight_in_location")
-      ind_df <- sub_df[, select_cols, with=FALSE][!is.na(sub_df[[var_weight]]) & !is.na(sub_df[[metric]]), ]
+      ind_df <- sub_df[!is.na(sub_df[[var_weight]]) & !is.na(sub_df[[metric]]), select_cols, with=FALSE]
```

Comment:
```suggestion
      ind_df <- sub_df[!is.na(sub_df[[var_weight]]) & !is.na(sub_df[[metric]]), ..select_cols]
```

I was wondering what `with=FALSE` does and I think you can do this instead, according to data.table's docs: https://rdatatable.gitlab.io/data.table/reference/data.table.html

It even says that "Note that `x[, cols]` is equivalent to `x[, ..cols]` and to x`[, cols, with=FALSE]` and to `x[, .SD, .SDcols=cols]`", so maybe `..` isn't needed either. This appears to be true as of data.table [v1.11.0](https://rdatatable.gitlab.io/data.table/news/index.html#data-table-v1-11-0-01-may-2018-2018-05-01) (see item 18 under New Features).

Reply:
From a small benchmarking experiment, the `with` format appears to be faster (~16%) than `..`. The New Features note does say they intend to deprecate `with` in "the next few years" (as of 2018) in favor of the dots, so maybe now's the time to switch?

Reply:
I was definitely not expecting a performance difference there...

Reviewing the data.table release notes, they generally seem to communicate deprecations well in advance. First new versions will issue a warning to say the feature is deprecated, and a year or two later they'll remove it. So we're probably safe to benefit from the performance gain for now. Maybe it'd be worth adding a comment to the code so if they do deprecate it a year from now, we remember why we used `with`.